src/draggable.ts

```diff
@@ -42,27 +57,88 @@ export class Draggable extends EventTarget {
       options: DraggableOptions = {}) {
     super();
 
+    this.over = {tag: 'NonTarget'};
+
     this.options = applyDefaults(options, {moveX: true, moveY: true});
+    this.$targets = this.options.$targets;
     this.setDimensions($parent);
 
-    let startPosn: Point;
     slide($el, {
       start: () => {
         if (this.disabled) return;
-        startPosn = this.position;
+        this.startPos = this.position;
         this.trigger('start');
         $html.addClass('grabbing');
       },
       move: (posn, start) => {
         if (this.disabled) return;
-        this.setPosition(startPosn!.x + posn.x - start.x,
-                         startPosn!.y + posn.y - start.y);
-        this.trigger('drag', this.position);
+        this.setPosition(this.startPos.x + posn.x - start.x,
+            this.startPos.y + posn.y - start.y);
+        this.trigger('drag', {elemPos: this.position, pointerPos: posn});
+
+        let overTarget = false;
+        const prevTarget = this.over.tag == 'NonTarget' ? undefined : this.over.$el;
+        if (this.$targets != undefined) {
```

Comment:
Make sue you always use `===` and `!==` rather than `==` and `!=`. In this case, just `if (this.$targets) {` should be enough